### PR TITLE
[v5.0.1] EN-2195 add edit file

### PR DESCRIPTION
**Pull request by fedelocci**

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

File: sass/file-browser/common/CreateTextFileForm.scss [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
   &__btn-cancel {
     margin-right: 2em;
-}
+  }
 
-}
+  &__download-filename-icon {
+    margin-left: 5px;
+  }
+}
```
Comment (nicpuddu) [inline sass/file-browser/common/CreateTextFileForm.scss]: missing end of file new line

File: sass/index.scss [hunk 2]
```diff
@@ -83,7 +82,7 @@
 @import "file-browser/list/FilesListTable";
 @import "file-browser/list/FileBreadcrumb";
 @import "file-browser/upload/UploadFileBrowserForm";
-@import "file-browser/add/CreateTextFileForm";
+@import "file-browser/common/CreateTextFileForm";
 @import "activity-stream/Notification";
 @import "profile-types/common/ProfileTypeForm";
-@import "profile-types/list/ProfileTypeListTable";
+@import "profile-types/list/ProfileTypeListTable";
```
Comment (nicpuddu) [inline sass/index.scss]: same here

File: src/ui/file-browser/edit/EditTextFileFormContainer.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import { getParams } from '@entando/router';
+import { download } from 'ui/file-browser/utils/downloadFile';
+import { saveFile, downloadFile, fetchFile } from 'state/file-browser/actions';
+
+import CreateTextFileForm from 'ui/file-browser/common/CreateTextFileForm';
+
+export const mapStateToProps = state => ({
+  mode: 'edit',
+  initialValues: {
+    extension: '.txt',
```
Comment (nicpuddu) [inline src/ui/file-browser/edit/EditTextFileFormContainer.js]: why does extension have an initial value of txt? what if it is a css? if the value is being retrieved by the onWillMount what is the point of the default?

File: src/ui/file-browser/common/CreateTextFileForm.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    onWillMount(filename);
+  }
+  render() {
+    const {
+      invalid, submitting, handleSubmit, mode, filename, onClickDownload,
+    } = this.props;
+    const onSubmit = (ev) => {
+      ev.preventDefault();
+      handleSubmit();
+    };
```
Comment (nicpuddu) [inline src/ui/file-browser/common/CreateTextFileForm.js]: this is beyond unnecessary: handlesubmit already prevents the default. you should just have handle submit as the form onSubmit property:

` <form onSubmit={handleSubmit} >` there is no need for any of this additional stuff.

File: src/ui/file-browser/common/CreateTextFileForm.js [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    </div>
+  );
+};
+
+
+export class CreateTextFileFormBody extends Component {
+  componentWillMount() {
+    const { filename, onWillMount } = this.props;
+    onWillMount(filename);
+  }
```
Comment (nicpuddu) [inline src/ui/file-browser/common/CreateTextFileForm.js]: it seems to me that there is no any actual check on this to see whether or not is a txt or css file. This basically means that if someone uses the correct url they can edit a non allowed file

File: src/state/file-browser/actions.js [hunk 6]
```diff
@@ -34,6 +35,42 @@ const wrapApiCall = apiFunc => (...args) => async (dispatch) => {
...
+            dispatch(initialize('CreateTextFileForm', { content: window.atob(json.payload.base64) }));
+          } else {
+            dispatch(addErrors(json.errors.map(e => e.message)));
+          }
+          dispatch(toggleLoading('file'));
+          resolve();
+        });
+      });
+    } else {
+      dispatch(addToast('Error file extension'), TOAST_ERROR);
```
Comment (nicpuddu) [inline src/state/file-browser/actions.js]: are we still displaying the form to edit the file at this point? if that is the case we should probably redirect or give the means to redirect.

**Comment by entando-service** [comment]
All is well

**Pull request merged**